internal/clients/nightfall/nightfall_test.go

```diff
@@ -134,13 +129,72 @@ func (n *nightfallTestSuite) TestScan() {
 		},
 		{},
 	}
+	expectedTooManyRequestsHTTPResponse := &http.Response{StatusCode: http.StatusTooManyRequests}
+
+	tests := []struct {
+		haveSuccess        bool
+		haveNumReqAttempts int
+		wantResponse       [][]nightfallAPI.ScanResponse
+		desc               string
+	}{
+		{
+			haveSuccess:        true,
+			haveNumReqAttempts: 1,
+			wantResponse:       expectedScanResp,
+			desc:               "success on first scan request",
+		},
+		/*{
+			haveSuccess:        true,
+			haveNumReqAttempts: 3,
+			wantResponse:       expectedScanResp,
+			desc:               "success on third scan request attempt",
+		},
+		{
+			haveSuccess:        false,
+			haveNumReqAttempts: 5,
+			wantResponse: nil,
+			desc: "failed after max retries",
+		},*/
+	}
+	for _, tt := range tests {
+		mockAPIClient := nightfallapi_mock.NewNightfallAPI(ctrl)
+		mockScanAPI := nightfallscanapi_mock.NewNightfallScanAPI(ctrl)
+		client := nightfall.Client{
+			APIClient:       mockAPIClient,
+			DetectorConfigs: detectorConfig,
+		}
 
-	mockAPIClient.EXPECT().ScanAPI().Return(mockScanAPI)
-	mockScanAPI.EXPECT().ScanPayload(gomock.Any(), gomock.AssignableToTypeOf(expectedScanReq)).Return(expectedScanResp, nil, nil)
+		numRetries := tt.haveNumReqAttempts
+		if tt.haveSuccess {
+			numRetries--
+		}
+		mockAPIClient.EXPECT().ScanAPI().Return(mockScanAPI)
+		// TODO find out how to send back 429 responses for initial tries
```

Comment:
running into the same issue as `GetRawBySha`, where the http client is nested in the `APIClient`